extensions/wikia/UserProfilePageV3/Masthead.class.php

```diff
@@ -447,23 +454,52 @@ public function isDefault() {
 	 * @return bool result
 	 */
 	private function doRemoveFile() {
-		wfProfileIn(__METHOD__);
+		wfProfileIn( __METHOD__ );
 
 		global $wgAvatarsUseSwiftStorage;
 		if ( !empty( $wgAvatarsUseSwiftStorage ) ) {
 			$swift = $this->getSwiftStorage();
 
 			$avatarRemotePath = $this->getLocalPath();
-			$status = $swift->remove($avatarRemotePath);
+			$status = $swift->remove( $avatarRemotePath );
 
 			$res = $status->isOk();
-		}
-		else {
+		} else {
 			$sImageFull = $this->getFullPath();
 			$res = file_exists( $sImageFull ) && unlink( $sImageFull );
 		}
 
-		wfProfileOut(__METHOD__);
+		if ( $res ) {
+			// remove thumbnails
+			$this->purgeThumbnails();
+		}
+
+		wfProfileOut( __METHOD__ );
+
+		return $res;
+	}
+
+	/**
+	 * Check if avatar file exists
+	 *
+	 * @return bool
+	 */
+	public function fileExists() {
+		wfProfileIn( __METHOD__ );
```

Comment:
I added it to because wanted to avoid creating the only method in this class that don't use MW profiler. 

On the other hand while we don't use it, maybe we should completely retire those methods? @wladekb what do you think?